src/servicesHost.ts

```diff
@@ -187,8 +189,11 @@ function makeResolvers(
   ): (typescript.ResolvedTypeReferenceDirective | undefined)[] =>
     typeDirectiveNames.map(
       directive =>
-        resolveTypeReferenceDirective(directive, containingFile)
-          .resolvedTypeReferenceDirective
+        resolveTypeReferenceDirective(
+          directive,
+          containingFile,
+          _redirectedReference
```

Comment:
This wasn't related to the particular repro provided, but it seemed like a place that might be broken for even edgier edge cases